File: leetcode/1436.py
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        c = set()
        s = set()
        for path in paths:
            c.add(path[0])
            c.add(path[1])
            s.add(path[0])
        d = c - s
        return d.pop()

File: leetcode/1436.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    string destCity(vector<vector<string>> &paths)
    {
        set<string> startingCities;
        for (vector<string> path : paths)
        {
            startingCities.insert(path[0]);
        }
        for (vector<string> path : paths)
        {
            if (startingCities.find(path[1]) == startingCities.end())
            {
                return path[1];
            }
        }
        return "";
    }
};

int main()
{
    Solution s = Solution();
    // assert();
}